### DISCUSSION TITLE
Is there a way to get this addon to work on third party twitch sites?

### DISCUSSION THREAD

**Discussion by kurtdh** [younesaassila/ttv-lol-pro#94] [Q&A]
Sites like lazygoat.tv, multitwitch.live, etc. still show ads when this add on is enabled.

The addon seems to work great at twitch.tv but doesn't work on any of those other third party sites. Is there any way to get this addon to work for those sites as well?

**Comment by HackerPide** [comment]
You can edit the `TWITCH_URL_REGEX` variable in https://github.com/younesaassila/ttv-lol-pro/blob/main/src/common/ts/regexes.ts#L2 to get the addon to work on more sites.

**Reply by kurtdh** [reply]
Thanks for the help. Unfortunately, I'm not that technically savvy. I click the xpi file and install it into Firefox as an extension. I also don't know that much about REGEX, how to add multiple URL's into that .ts file, and then what to do with that .ts file once I'm done editing it. Do I have to convert the .ts file into an .xpi file before I can install it into Firefox? 

I would absolutely LOVE this capability but I'm afraid my technical side isn't good enough to know how to do what you're explaining.

**Comment by younesaassila** [comment]
Hey, the proxying part of the add-on is entirely feasible: this would just require 2 additional permissions for the add-on (`player.twitch.tv` and `embed.twitch.tv`). The "stream status"/"Reset player on midroll" features however would be way more complex since both currently rely on the Twitch.tv website directly.

**Comment by Equ1no0x** [comment]
You can just modify the manifest.json to have them in the "matches" section. I have yet to test with `https://multitwitch.live/` or `https://kadgar.net/live` which are the multiviewers I usually use.

`"matches": ["https://www.twitch.tv/*", "https://m.twitch.tv/*", "https://player.twitch.tv/*", "https://embed.twitch.tv/*"]`

**Reply by kurtdh** [reply]
Must I also edit the regexes.ts file like hackerpride said above, or will this be all I have to do?

**Reply by Equ1no0x** [reply]
> Must I also edit the regexes.ts file like hackerpride said above, or will this be all I have to do?

I didn't do that, just manifest.json

**Reply by HackerPide** [reply]
> Must I also edit the regexes.ts file like hackerpride said above, or will this be all I have to do?

Sorry for the confusion. I assumed that step would be needed as it was done when adding support for m.twitch.tv .

**Reply by dmmdanielg** [reply]
Did you have any problems editing that file? I tried editing the v1.8.1-unpacked and that didn't get me an error but didn't work.  It registers in chrome but when I click the extension in the extension bar it doesn't recognize a stream to block. I can clearly see it names the stream I'm watching when I use native twitch.tv.

https://www.destiny.gg/bigscreen#twitch/[stream_name] I'm trying to get this to work but no luck.

**Reply by Equ1no0x** [reply]
> "https://player.twitch.tv/*"

Try "https://www.destiny.gg/bigscreen#twitch/*"
